src/snapred/backend/dao/indexing/Record.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any
+
+from pydantic import ConfigDict, field_validator
+from snapred.backend.dao.indexing.CalculationParameters import CalculationParameters
+from snapred.backend.dao.indexing.IndexEntry import IndexEntry
+from snapred.backend.dao.indexing.Versioning import VersionedObject
+
+
+class Record(VersionedObject, extra="allow"):
+    """
+
+    This is the basic, bare-bones record of a workflow completion.
+    It contains only the run number, the resolution (native/lite), and a version.
+
+    This is meant to coordinate with the Indexer service object.
+
+    The class method `indexFromRecord` will create a compatible index entry from a record.
+
+    """
+
+    # inherits from VersionedObject
+    # - version: int
+
+    runNumber: str
+    useLiteMode: bool
+    # NOTE calculationParameters is a VERSIONED object.
+    # the version on the calculation parameters MUST match the version on this record.
+    # a future validator should enforce this condition
+    calculationParameters: CalculationParameters
+
+    @classmethod
+    def indexEntryFromRecord(cls, record) -> IndexEntry:
+        entry = None
+        if record is not None:
+            entry = IndexEntry(
+                runNumber=record.runNumber,
+                useLiteMode=record.useLiteMode,
+                version=record.version,
+                appliesTo=f">={record.runNumber}",
```

Comment:
The way this is used would create an arbitrary NEW entry in the index, granted with minimal priority, but Im not sure we want to try and generically infer an entry